crates/ethereum_bridge/src/protocol/transactions/ethereum_events/events.rs

```diff
@@ -162,48 +162,9 @@ where
     let native_werc20_supply_key =
         minted_balance_key(&erc20_token_address(native_erc20));
 
-    update::amount(state, &eth_bridge_native_token_balance_key, |balance| {
-        tracing::debug!(
-            %eth_bridge_native_token_balance_key,
-            ?balance,
-            "Existing value found",
-        );
-        balance.spend(amount)?;
-        tracing::debug!(
-            %eth_bridge_native_token_balance_key,
-            ?balance,
-            "New value calculated",
-        );
-        Ok(())
-    })?;
-    update::amount(state, &receiver_native_token_balance_key, |balance| {
-        tracing::debug!(
-            %receiver_native_token_balance_key,
-            ?balance,
-            "Existing value found",
-        );
-        balance.receive(amount)?;
-        tracing::debug!(
-            %receiver_native_token_balance_key,
-            ?balance,
-            "New value calculated",
-        );
-        Ok(())
-    })?;
-    update::amount(state, &native_werc20_supply_key, |balance| {
-        tracing::debug!(
-            %native_werc20_supply_key,
-            ?balance,
-            "Existing value found",
-        );
-        balance.spend(amount)?;
-        tracing::debug!(
-            %native_werc20_supply_key,
-            ?balance,
-            "New value calculated",
-        );
-        Ok(())
-    })?;
+    let native_token = state.in_mem().native_token.clone();
+    token::transfer(state, &native_token, &BRIDGE_ADDRESS, receiver, *amount)?;
+    decrement_total_supply(state, &erc20_token_address(native_erc20), *amount)?;
```

Comment:
I think the storage key this is updating is missing from the returned `BTreeSet` of changed keys, no?

Reply:
In the diff, I didn't remove any storage keys, and the returned `BTreeSet` still contains three storage keys changed, which I think is the right number. Unless you think there is a bug that existed before my changes?

Reply:
nvm, the affected key is `native_werc20_supply_key` which is def in the returned sed